telegram/tests/test_telegram.py

```diff
@@ -0,0 +1,344 @@
+import json
+
+from twisted.internet.defer import inlineCallbacks, returnValue, DeferredQueue
+from twisted.web.server import NOT_DONE_YET
+from twisted.web import http
+
+from vumi.tests.utils import LogCatcher
+from vumi.tests.helpers import VumiTestCase
+from vumi.tests.fake_connection import FakeHttpServer
+from vumi.transports.httprpc.tests.helpers import HttpRpcTransportHelper
+
+from telegram.telegram import TelegramTransport
+
+
+class TestTelegramTransport(VumiTestCase):
+
+    @inlineCallbacks
+    def setUp(self):
+        self.helper = self.add_helper(
+            HttpRpcTransportHelper(TelegramTransport)
+        )
+        self.request_queue = DeferredQueue()
+        self.pending_requests = []
+        self.addCleanup(self.finish_requests)
+        self.mock_server = FakeHttpServer(self.handle_inbound_request)
+
+        self.default_user = {
+            'id': 'default_user_id',
+            'username': '@default_user',
+        }
+        self.bad_telegram_response = {
+            'ok': False,
+            'description': 'Bad request',
+        }
+        self.bot_username = '@bot'
+        self.API_URL = 'https://api.telegram.org/bot'
+        self.TOKEN = '1234'
+
+        self.transport = yield self.get_transport()
+
+        # Telegram chat types
+        self.PRIVATE = 'private'
+        self.CHANNEL = 'channel'
+        self.GROUP = 'group'
+
+    @inlineCallbacks
+    def get_transport(self, **config):
+        defaults = {
+            'bot_username': self.bot_username,
+            'bot_token': self.TOKEN,
+            'web_path': 'foo',
+            'web_port': 0,
+            'inbound_url': 'www.example.com',
+            'outbound_url': self.API_URL
+        }
+        defaults.update(config)
+        transport = yield self.helper.get_transport(defaults)
+        transport.agent_factory = self.mock_server.get_agent
+        returnValue(transport)
+
+    def handle_inbound_request(self, req):
+        self.request_queue.put(req)
+        return NOT_DONE_YET
+
+    @inlineCallbacks
+    def get_next_request(self):
+        req = yield self.request_queue.get()
+        self.pending_requests.append(req)
+        returnValue(req)
+
+    @inlineCallbacks
+    def finish_requests(self):
+        for req in self.pending_requests:
+            if not req.finished:
+                yield req.finish()
+
+    def test_get_outbound_url(self):
+        test_url = self.transport.get_outbound_url('myPath')
+        self.assertEqual(test_url, '%s%s/%s' %
+                         (self.API_URL, self.TOKEN, 'myPath'))
```

Comment:
Each test method should have a descriptive docstring that explains what is being tested. For this method, the doctring might be something like "The transport correctly builds an outbound URL for a given path."

The idea is that the docstring tells you what the purpose of the test is so that someone trying to understand it later (because it failed when they changed some code) can see what the test is supposed to be testing and then either fix their code (if the test failure points to a bug), fix the test (if the test has a bug), or update the test (if the desired behaviour has changed). You can't necessarily rely on the test code for this, because the code doesn't necessarily test the thing it's supposed to be testing, or it include a bunch of setup and such that obscure its intent.